File: LeetCodeCPP/338.counting-bits.cpp
/*
 * @lc app=leetcode id=338 lang=cpp
 *
 * [338] Counting Bits
 */

#include <vector>

using namespace std;

// @lc code=start
class Solution
{
public:
    vector<int> countBits(int n)
    {
        vector<int> bits(n + 1, 0);

        for (size_t i = 1; i <= n; i++)
        {
            bits[i] = 1 + bits[i & (i - 1)];
        }
        return bits;
    }
};
// @lc code=end

File: LeetCodePython/338.counting-bits.py
#
# @lc app=leetcode id=338 lang=python3
#
# [338] Counting Bits
#

from typing import List


# @lc code=start
class Solution:
    def countBits(self, n: int) -> List[int]:
        bits = [0] * (n + 1)

        for i in range(1, n + 1):
            bits[i] = 1 + bits[i & (i - 1)]

        return bits


# @lc code=end
